ouroboros-consensus/src/Ouroboros/Consensus/Storage/ImmutableDB/Impl/State.hs

```diff
@@ -177,15 +187,15 @@ modifyOpenState
   -> (  forall h.
         TempResourceRegistry m
      -> HasFS m h
-     -> StateT (OpenState m hash h) m r
+     -> StateT (OpenState m hash h) m ([ResourceKey m], r)
```

Comment:
Use https://hackage.haskell.org/package/transformers-0.5.6.2/docs/Control-Monad-Trans-RWS-CPS.html#t:RWST with `TempResourceRegistry m` as R. Call it `ModifyOpenState m hash h`.